R/functions_io.R

```diff
@@ -240,6 +241,12 @@ ReadSparseMatrix = function(path, project="SeuratProject", row_name_column=2, co
   }
 
   # special case: if hashtags are specified and if there is a Antibody Capture data, then split and add as separate assay
+  # if a regular expression is defined for HTO, overwrite hto_names 
+  if ("Antibody Capture" %in% names(feature_data) && !is.null(hto_regex) && nchar(hto_regex)>0) {
+    hto_names = grep(pattern=hto_regex, x=rownames(feature_data[["Antibody Capture"]]), v=TRUE)
+    if (length(hto_names)==0) futile.logger::flog.error("Could not find HTO names with the 'hto_regex' argument '%s'!", hto_regex)
```

Comment:
So ... I think we said some time ago, we drop the `futile.logger`?

Reply:
I replaced it with `stop(sprintf("Could not find HTO names with the 'hto_regex' argument '%s'!", hto_regex))`. Use of `sprintf` makes it more readable then `paste`.

Reply:
Why not use a function like we use for messages and warnings?

Reply:
Can we take care of this before the release - i mean it is not really related to HTO but something more general?

Reply:
It came up due to a change you introduced in the hto branch... I will note it down, we'll take care of it later.